bookkeeper-server/src/main/java/org/apache/bookkeeper/conf/ServerConfiguration.java

```diff
@@ -4102,4 +4109,113 @@ public ServerConfiguration setLedgerMetadataRocksdbConf(String ledgerMetadataRoc
         this.setProperty(LEDGER_METADATA_ROCKSDB_CONF, ledgerMetadataRocksdbConf);
         return this;
     }
+
+    /**
+     * Returns the number of bookies to run in this cluster.
+     *
+     * @return the number of bookies to run in this cluster.
+     */
+    public int getNumBookies() {
+        return getInt(TABLE_NUM_BOOKIES, 1);
+    }
+
+    /**
+     * Set the number of bookies to run in this cluster.
+     *
+     * @param numBookies the number of bookies
+     * @return ServerConfiguration
+     */
+    public ServerConfiguration setNumBookies(int numBookies) {
+        setProperty(TABLE_NUM_BOOKIES, numBookies);
+        return this;
+    }
+
+    /**
+     * Returns if should start zookeeper.
+     *
+     * @return true if should start zookeeper, otherwise false.
+     */
+    public boolean getShouldStartZooKeeper() throws ConfigurationException {
+        return getMetadataServiceUri() == null;
+    }
+
+    /**
+     * Returns the zookeeper server port used in this cluster.
+     *
+     * @return the zookeeper server port used in this cluster.
+     */
+    public int getZKPort() {
```

Comment:
The table service shares the same Zookeeper with the Bookie, why do we set the zookeeper port?